src/script/lua_api/l_util.cpp

```diff
@@ -398,7 +398,8 @@ int ModApiUtil::l_get_dir_list(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	const char *path = luaL_checkstring(L, 1);
-	short is_dir = lua_isboolean(L, 2) ? lua_toboolean(L, 2) : -1;
+	bool list_all = lua_type(L, 2) != LUA_TBOOLEAN; // if its not a boolean list all
```

Comment:
Well lua_isboolean is a macro and not type save.
`!lua_isboolean(L, 2)` would resolved as `!(lua_type(L, (n)) == LUA_TBOOLEAN)` which would be a bit stupid imo. Maybe it gets optimized by the compiler, so it shouldn't make a difference at all.
If you think `!lua_isboolean(L, 2) `is better this could be changed while merging.